src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupDebug.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+
+namespace Internal.Runtime.CompilerHelpers
+{
+    /// <summary>
+    /// <see cref="System.Diagnostics.Debug"/> replacement for the startup path.
+    /// It's not safe to use the full-blown Debug class during startup because big chunks
+    /// of the managed execution environment are not initialized yet.
+    /// </summary>
+    static class StartupDebug
+    {
+        [Conditional("DEBUG")]
+        public static void Assert(bool condition)
```

Comment:
What would it buy us? It would make Debug.cs more complicated and won't lead to any code sharing because the only purpose of the flag would be to avoid running any of the existing code.